src/modules/GroupChannelList/context/GroupChannelListProvider.tsx

```diff
@@ -222,3 +206,36 @@ export function useGroupChannelListContext(): GroupChannelListProviderInterface
   const context: GroupChannelListProviderInterface = useContext(GroupChannelListContext);
   return context;
 }
+
+function getCollectionCreator(sdk: SdkStore['sdk'], channelListQuery?: GroupChannelListQueryParamsInternal) {
+  if (!channelListQuery) return undefined;
+
+  return () => {
+    const filter = new GroupChannelFilter();
+
+    filter.includeEmpty = channelListQuery.includeEmpty ?? false; // uikit default: false
+    filter.includeFrozen = channelListQuery.includeFrozen ?? filter.includeFrozen;
+    filter.nicknameContainsFilter = channelListQuery.nicknameContainsFilter ?? filter.nicknameContainsFilter;
+    filter.channelNameContainsFilter = channelListQuery.channelNameContainsFilter ?? filter.channelNameContainsFilter;
+    filter.customTypesFilter = channelListQuery.customTypesFilter ?? filter.customTypesFilter;
+    filter.customTypeStartsWithFilter = channelListQuery.customTypeStartsWithFilter ?? filter.customTypeStartsWithFilter;
+    filter.channelUrlsFilter = channelListQuery.channelUrlsFilter ?? filter.channelUrlsFilter;
+    filter.superChannelFilter = channelListQuery.superChannelFilter ?? filter.superChannelFilter;
+    filter.publicChannelFilter = channelListQuery.publicChannelFilter ?? filter.publicChannelFilter;
+    filter.myMemberStateFilter = channelListQuery.memberStateFilter ?? filter.myMemberStateFilter;
+    filter.hiddenChannelFilter = channelListQuery.hiddenChannelFilter ?? filter.hiddenChannelFilter;
+    filter.unreadChannelFilter = channelListQuery.unreadChannelFilter ?? filter.unreadChannelFilter;
+
+    if (typeof channelListQuery.userIdsExactFilter !== 'undefined') {
```

Comment:
might be better to replace with
```
if (!channelListQuery.userIdsExactFilter) {
```
which could cover null as well

Reply:
Good!